projects/packages/search/src/instant-search/components/overlay.jsx

```diff
@@ -14,6 +14,30 @@ const Overlay = props => {
 			}
 		};
 
+		const handleTabEvent = event => {
+			if ( event.key === 'Tab' ) {
+				const overlay = document.getElementsByClassName( OVERLAY_CLASS_NAME )[ 0 ];
+				const isInsideOverlay = overlay.contains( event.target );
+
+				// Looking up the searchInput assumes knowledge of another component.
+				// Not currently a constent as it's dynamically generated.
+				const searchInput = document.getElementById( 'jetpack-instant-search__box-input-1' );
+				const tabAnchor = document.getElementById( OVERLAY_FOCUS_ANCHOR_ID );
+				if ( event.shiftKey === true ) {
+					if ( event.target === searchInput || false === isInsideOverlay ) {
+						event.preventDefault();
+						tabAnchor.focus();
+					}
+				}
+				if ( event.shiftKey === false ) {
+					if ( event.target === tabAnchor || false === isInsideOverlay ) {
+						event.preventDefault();
+						searchInput.focus();
+					}
+				}
```

Comment:
##### Do you think the logic would be a bit clearer this way? 👆

##### Dynamic ID

> // Not currently a constant as it's dynamically generated.

I guess it might not be the best idea to rely on a dynamically generated value, because a refactoring would easily break this.

Possible solutions:

- Could we maybe change the dynamic ID to a constant one in the overlay (I don't see any particular reason why it's not)
- Or we could use a custom attribute to identify the input, like use a data attribute? We could then query the element with class + the attribute.

`<input data-focus-trap="start" class="jetpack-instant-search__box-input" ... />`

- Or the least we could do is to add a comment back in there, saying it's relied on for tab trapping.

Personally I would prefer the first choice 😄 Or you have any other idea?

##### Naming
Lastly, do you think it's better to name the variables like `startAnchorOfFocusTrapping` and `endAnchorOfFocusTrapping` (you probably have better ones) which would better explain why we are focusing on them on the conditions.

Reply:
These suggested changes won't capture Tab events outside the overlay. Part of the logic is to make sure tabbing from the browser Chrome into the DOM is also captured. As an example, if you focus the address bar and then hit Tab, it will move to the first item on the page. The second Tab will be visible to us and we use that to refocus into the overlay.

> I guess it might not be the best idea to rely on a dynamically generated value, because a refactoring would easily break this.

Yes, I wasn't sure how best to address this. I'll do some testing with your suggestions. Option one sounds like the way to go.

> Lastly, do you think it's better to name the variables like […]

Yeah, I think you're right. I'll revisit this too.

Reply:
> These suggested changes won't capture Tab events outside the overlay. 

Ah you're right. All good then 👍

Reply:
Hmmm do we really want to trap focus even focus is outside the overlay? 🙋‍♂️

Reply:
> Hmmm do we really want to trap focus even focus is outside the overlay? 🙋‍♂️

Yup! If not, the user has to tab though all the underlying obscured page links to get back into the overlay. The focus-trap is only in effect when the overlay is visible. From [the original issue](https://github.com/Automattic/jetpack/issues/23396):

> It is possible for a keyboard user to tab out of the search modal without closing it. Once this happens, there is no visual reference for the current location of focus, and it may be very difficult to return to the modal.

> While the modal is open, keyboard focus should be limited to only allow access to focusable elements that are inside the modal.

> See WCAG 2.4.3: https://www.w3.org/TR/UNDERSTANDING-WCAG20/navigation-mechanisms-focus-order.html

Those underlying tab targets should be inaccessible until the overlay is dismissed.